airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/backward_compatibility.py

```diff
@@ -0,0 +1,155 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import jsonschema
+from airbyte_cdk.models import ConnectorSpecification
+from deepdiff import DeepDiff
+from hypothesis import given, settings
+from hypothesis_jsonschema import from_schema
+from source_acceptance_test.utils import SecretDict
+
+
+class NonBackwardCompatibleSpecError(Exception):
+    pass
+
+
+class SpecDiffChecker:
+    """A class to perform multiple backward compatible checks on a spec diff"""
+
+    def __init__(self, diff: DeepDiff) -> None:
+        self._diff = diff
+
+    def check(self):
```

Comment:
nit: I would rename this to something like `assert_specs_are_backwards_compatible` or something so it is clear in the calling context this is an assertion.

Reply:
renamed to `assert_spec_is_backward_compatible`